### ISSUE TITLE
What does the output of onsets_frames_wavinput.tflite mean?

### ISSUE THREAD

**Issue by WonyJeong** [magenta/magenta#1943]
Hello,

I am developing a piano education application using a flutter.

I'm a front-end developer and my knowledge of machine learning is not very deep.

Currently in my App, Can successfully load model and return outputs of onsets_frames_wavinput.tflite on Android/iOS.

Here some code (android)

``` java
      Object[] inputArray = {floatInputBuffer}; // size : 17920
      Map<Integer, Object> outputMap = new HashMap<>();
      float[][][] floatOutputBuffer = new float[1][32][88];
      outputMap.put(0, floatOutputBuffer);

      tfLite.runForMultipleInputsOutputs(inputArray,outputMap);  // The outputMap is an array of size 1x32x88.
```

88 seems to mean the number of piano keys. **But I don't know what the 32 in the array means.**
Does 32 mean that the input buffer was processed for 32ms?

A lot of people seem to be wondering how to utilize this model in mobile applications.

I want to create an open source that successfully processes output and share it with people.

**Comment by WonyJeong** [comment]
I read the paper and got an understanding of the output of onsets_frames_wavinput.tflite.
And by processing onset and frame, the code for recognizing the piano keyboard was completed.

I have distributed a flutter plugin applicable to iOS and Android, except for the method channel code for communication between flutter and native, it can be applied to react-native, kotlin, and swift as well.

Here is the link.

https://pub.dev/packages/flutter_piano_audio_detection

**Comment by cghawthorne** [comment]
Wow, that's great! Do you have a demo we could link to?

**Comment by WonyJeong** [comment]
> Wow, that's great! Do you have a demo we could link to?

I haven't made a demo yet.

I'll comment when a simple working demo video (or image) is ready!

**Comment by WonyJeong** [comment]
HI @cghawthorne ! 

I have one question for you.
Is it okay to send the PR that I uploaded the flutter plugin link that I developed on /onsets_frames_transcription/realtime/readme ?

**Comment by WonyJeong** [comment]
@AlanPaine  nope, I didn't train this model myself. My work here was all about integrating the *already trained* TFLite model into the app and making sure the real-time inference worked okay.
